### PR TITLE
refactor/GRW-602-integrate QuoteCart API on Campaign code management 

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## What?

Following the same pattern used during the whole _integrate QuoteCart API_ work, I'm making the changes by adding cloning the components that would be affected by the transition and change them so they use the new API. In this case, the cloned components are: 

`src/client/pages/OfferNew/Introduction/index.tsx` --> `src/client/pages/OfferNew/Introduction/Introduction.tsx`
`src/client/pages/OfferNew/Introduction/Sidebar/index.tsx` --> `src/client/pages/OfferNew/Introduction/Sidebar/Sidebar.tsx`
`src/client/pages/OfferNew/Introduction/Sidebar/DiscountCodeModal.tsx` --> `src/client/pages/OfferNew/Introduction/Sidebar/CampaignCodeModal.tsx`

### The changes

- `Introduction` and `Sidebar` now accepts a `quoteCartId`. `Introduction` gets it from `Offer` page and then passes it down to `Sidebar` who, among other things, uses it to get and render  `campaign` info using `QuoteCart API`
- `Sidebar` make usage of mutations to add/remove campaign codes. Those mutations are part of the new API.
- `Sidebar` renders a `CampaignCodeModal` component and provide to it a callback to be called during the attempt of adding a campaign code.
- Some renaming were made as requested. So _campaign (code)_ term is being used rather than _discount/redeem_.

### Todo

- [x] fix currency/locale bug
- [x] add loading state to _Add Campaign_ form
- [ ] rename lokalise keys: _Redeem/Discount_ --> _Campaign (code)_

## Why?

- The new API should be used on Campaign during code management 

**Ticket(s): [GRW-602](https://hedvig.atlassian.net/browse/GRW-602)**

## Dependencies

#705 - It will be impacted by this one since it changes `DiscountTag` inner works

### CODE REVIEW

**Review by robinandeer — CHANGES_REQUESTED** [review]
Let's avoid mixing existing and quote cart API components.

Prefer putting the new `Sidebar`/`Introduction` under `Offer` instead of `OfferNew` :)

File: src/client/pages/Offer/index.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ import {
 import { AppPromotionSection } from '../OfferNew/AppPromotionSection'
 import { Checkout } from '../OfferNew/Checkout'
 import { FaqSection } from '../OfferNew/FaqSection'
-import { Introduction } from '../OfferNew/Introduction'
+import { Introduction } from '../OfferNew/Introduction/Introduction'
```
[the remark this answers is not in the document]
Comment (mehdielket) [reply src/client/pages/Offer/index.tsx:32]: You can take a look at my Details Modal PR for reference on how we approached duplicating components

[the remark this answers is not in the document]
Comment (guilhermespopolin) [reply src/client/pages/Offer/index.tsx:32]: Do you mean place new `Introduction` inside `Offer/Introduction/index.tsx`?

**Comment by guilhermespopolin** [comment]
I closing this one and opening a new one: #718 due a renaming of the branch.

**Pull request closed without merging**